crates/codegen/runtime/cargo/wasm/src/runtime/interface/cst.wit.jinja2

```diff
@@ -157,13 +163,19 @@ interface cst {
         /// Returns the current depth in the tree (i.e. number of ancestors).
         depth: func() -> u32;
 
-        /// Returns the list of ancestor nodes up to the root.
-        ancestors: func() -> list<nonterminal-node>;
+        /// Returns the list of child edges directly connected to this node.
+        children: func() -> list<edge>;
+        /// Returns an iterator over all descendants of the current node in pre-order traversal.
+        descendants: func() -> cursor-iterator;
+        /// Returns an iterator over all the remaining nodes in the current tree, moving in pre-order traversal, until the cursor is completed.
+        consume: func() -> cursor-iterator;
```

Comment:
IIUC, the semantics of the generated function are different from the Rust's native function since this implementation will clone before calling to `consume`. Does it make sense to still call it `consume` in the WIT interface?

Reply:
Good point. To keep things consistent, I renamed it to `remainingNodes()` and modified the comment.